client/layout/index.jsx

```diff
@@ -261,7 +262,7 @@ class Layout extends Component {
 				<QuerySites primaryAndRecent={ ! config.isEnabled( 'jetpack-cloud' ) } />
 				{ this.props.shouldQueryAllSites && <QuerySites allSites /> }
 				<QueryPreferences />
-				<QuerySiteFeatures siteId={ this.props.siteId } />
+				<QuerySiteFeatures siteIds={ this.props.selectedOrAllSites.map( ( site ) => site.ID ) } />
```

Comment:
That part is a bit concerning. Without a selected site, I'm getting hundreds of additional requests to fetch features for all of my sites now. Do we really want to do this for all sites? Should it be just for Jetpack/Atomic sites?

Reply:
> Do we really want to do this for all sites? Should it be just for Jetpack/Atomic sites?

For the feature on #62353 we'll trusting entirely on feature checks instead of Jetpack/Atomic flags.

Reply:
That sounds a bit discouraging for users with lots of sites though.

Have you considered creating a new endpoint that will retrieve the features of all sites in a single request? At the very least I think we should compare such an approach with the one suggested here.